_topics/en/op_checktemplateverify.md

```diff
@@ -131,30 +131,230 @@ see_also:
+  vbytes.
+
+- **OP_CAT + OP_CHECKSIGFROMSTACK:** the composition of the proposed
+  [OP_CAT][topic op_cat] and [OP_CHECKSIGFROMSTACK][topic
+  op_checksigfromstack] opcodes can emulate CTV, although this may
+  require significantly more vbytes.
+
+- **OP_TX and OP_TXHASH:** are [roughly][op_tx] equivalent
+  [proposals][op_txhash] to enable CTV features in a way that might be
+  easier to extend in the future.  Depending on the exact design, they
+  may require slightly more vbytes to use than CTV.
+
+- **Exploding keys:** this [proposal][dryja exploding keys] provides the
+  core feature of CTV: the ability for a transaction output to commit to
+  the outputs of the transaction that will spend it.  CTV accomplishes
+  this using a hash digest on the stack; exploding keys accomplishes it
+  using a [pay-to-contract][topic p2c]-like commitment to the hash
+  digest.  Exploding keys can, in theory, save a few vbytes compared to
+  CTV---although it also loses some of CTV's flexibility.
+
+## Criticisms of CTV
+
+Several criticisms of CTV have been covered by Optech:
+
+- **Not flexible enough:** CTV was deliberately designed to provide a
+  limited function that would be useful in a variety of constructions
+  but without enabling [recursive covenants][topic covenants].  However,
+  [some critics][corallo flexible] would [prefer][corallo costs] to see
+  a more general covenant solution.  They worry that many of the people
+  who ask for CTV now will not be satisfied with it.  They may later ask
+  for an improved construction that could render CTV unnecessary.  They
+  advocate instead for effort to be devoted to finding a solution that
+  will fully satisfy user desires.
+
+- **Not generic enough:** CTV is almost maximally efficient at allowing
+  an output to require it be spent to certain future outputs.  It
+  additionally provides some flexibility in the construction of that
+  spending transaction, mainly related to which inputs it includes.
+  However, its capabilities overlap to a significant degree with other
+  [covenant][topic covenants] proposals, so it has been
+  [suggested][oconnor composable] that it would be better to use
+  composable elements to replicate the features of CTV.  Those elements
+  may also be useful on their own, or they may be extensible in ways
+  that don't necessarily make sense for CTV.  This is the origin of
+  proposals such as `OP_TX` (described previously).
+
+- **Makes script harder to analyze:** CTV tracks information about the
+  executed opcode that precedes it as part of its protection against
+  recursive covenants.  This is abnormal among opcodes
+  (`OP_CODESEPARATOR` in its interaction with `OP_CHECKSIG` and variants
+  being the only current opcode we're aware of that requires similar
+  tracking).  At least one reviewer argues that dealing with the
+  abnormal state requirements of CTV [adds][oconnor complexity] "much
+  more complexity to anyone trying to formally model Script semantics".
```

Comment:
this was true for OP_COSHV and removed by the CTV proposal in response to the feedback. Just a normal script semantic opcode now. @roconnor-blockstream